### ISSUE TITLE
`Uncaught runtime errors: "Module build failed (from ./node_modules/react-scripts/node_modules/babel-loader/lib/index.js): Error: [BABEL] /app/src/index.js: Cannot find module '@babel/plugin-proposal-private-property-in-object' "`

### ISSUE THREAD

**Issue by kyle-mucerino** [kyle-mucerino/workout-app#1]
I've followed a couple different issues that look similar to the one I'm having, [#13189](https://github.com/facebook/create-react-app/issues/13189) and [#15655](https://github.com/babel/babel/issues/15655) but it seems that those builds crash while building. I am able to get a successful Heroku build and deployment, but then when launching the app I get `Uncaught runtime errors: "Module build failed (from ./node_modules/react-scripts/node_modules/babel-loader/lib/index.js): Error: [BABEL] /app/src/index.js: Cannot find module '@babel/plugin-proposal-private-property-in-object' "`

I tried upgrading my `"@babel/preset-env"` to versions `"^7.22.1"` and now `"^7.22.3"` and deleting instances of `"@babel/plugin-proposal-private-property-in-object"` as well as adding `"@babel/plugin-proposal-private-property-in-object"` to my `"dependencies"` in addition to `"devDependencies"`. I have also removed Node_modules and reinstalled to no avail.

**Comment by JLHwung** [comment]
I have cloned the repo and run `npm start` and `npm run build`. Both look good to me.

**Comment by JLHwung** [comment]
I am not very familiar with heroku. Could you tell me how to launch the app from heroku cli?

**Comment by kyle-mucerino** [comment]
`heroku open` should launch the app from the CLI

**Comment by JLHwung** [comment]
I just installed heroku 8.1.9:

```
npx heroku open
 ›   Error: Missing required flag:
 ›    -a, --app APP  app to run command against
```
What should I put on the `--app` argument? `./public`?

**Comment by kyle-mucerino** [comment]
I think it would be `-a ./public`

**Comment by JLHwung** [comment]
I don't have access to this heroku app. If both `npm run build` and `npm start` work, I would say please check your heroku app configs, looking for any build cache settings and try to purge them.

**Comment by kyle-mucerino** [comment]
Ok, I can add you as a collaborator on the project but will try the heroku app configs and cache settings purge and see if that works first. Thank you.

**Comment by kyle-mucerino** [comment]
Tried clearing the cache using `heroku builds:cache:purge` but still having the issue.

**Comment by JLHwung** [comment]
Can you run `shasum -a 256 package-lock.json` on heroku? You can add that to the build step and see if the heroku instance are using the same lock file as it is currently in the git.

**Comment by kyle-mucerino** [comment]
Is that a command or something I need to add to a section in my `package-lock` ? Sorry I tried googling it and couldn't really find anything explaining `shasum -a 256`

**Comment by JLHwung** [comment]
`shasum` is available on Linux and macOS: https://linux.die.net/man/1/shasum
You can also use `md5sum` if available. The idea is to print the hash of the lock file and determine if the one on heroku is same as the one in git.

**Comment by kyle-mucerino** [comment]
That's very cool, I did not know about that so thank you!! Unfortunately the hash is the same in heroku and git so the problem remains unsolved.

**Comment by JLHwung** [comment]
If the lock files are same but it generates different node_modules layout on heroku than your local environment, I suspect it is still a heroku issue. You can try disabling the node_modules cache behaviour and see if it works: https://devcenter.heroku.com/articles/nodejs-support#cache-behavior

**Comment by kyle-mucerino** [comment]
Nope, same runtime error. Successful build but still getting the '@babel/plugin-proposal-private-property-in-object' error. I may just have to try and switch from Heroku to AWS but I'd really like to figure this out.
